kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -226,7 +227,12 @@ public Transaction build(Engine engine) {
       newMetadata = columnMappingMetadata;
     }
 
-    /* ----- 5: Validate the metadata change ----- */
+    /* ----- 5: Update the METADATA with the new schema if applicable ----- */
```

Comment:
where are we assigning the physical name/ids for new columns? Step 4 is the one that does it and also updates the max field id in the metadata.configuration.

Reply:
+1 I think my intention was that we would add the schema to `newMetadata` in step 1 (see the TODO); And then the validation falls where it currently is :)

Reply:
also if this is the case and no tests are failing let's try to add a test to cover it. spooky if we break things and a test doesn't fail

Reply:
Yes it should be in step 1. That way we don't skip the icebergcompatv2 checks as well (e.g new schema could add the unsupported column type - variant )

Reply:
Yeah taking a look at this, but to be clear we *do not* want to assign physical name/ids for new columns in the updated schema. The expectation is that those are already set. But we do need to ensure compatv2 checks + max field ID is set in metadat

Reply:
Ah i see what you mean technically we'd be missing it for new tables

Reply:
Discussed offline, previously was expecting that every new field must've already had a defined ID/physical name but it makes more sense for this API to assign new IDs/names. That's more generic, and for use cases that callers want to keep the ID/name, we'll handle that.